### PR TITLE
Add new versions of py-quast

### PR DESCRIPTION
**Pull request by Pandapip1**
The old versions are significantly out of date

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-quast/package.py [hunk 1]
```diff
@@ -11,8 +11,12 @@ class PyQuast(PythonPackage):
...
-    url = "https://github.com/ablab/quast/archive/quast_4.6.1.tar.gz"
+    pypi = "quast/quast-5.2.0.tar.gz"
 
+    version("5.2.0", sha256="23649fbd93253c6da982c0b67d719f9262461deecdc6dffbd690b75dfd790ad7")
+    version("5.0.2", sha256="cdb8f83e20cc38f218ff7172b454280fcb1c7e2dff74e1f8618cacc53d46b48e")
+    version("5.0.1", sha256="b1e4443b6598b01faaefddfc0f06fb270414ed4bdaffd0ad9aa420bc0d07d8ad")
+    version("5.0.0", sha256="46bba247c7f92c2ccaca8c0abeab2a8d40a257a0cbe2fa0a4ffa981ca0267526")
     version("4.6.3", sha256="d7f5e670563d17d683f6df057086f7b816b6a088266c6270f7114a1406aaab63")
     version("4.6.1", sha256="a8071188545710e5c0806eac612daaabde9f730819df2c44be3ffa9317b76a58")
     version("4.6.0", sha256="6bee86654b457a981718a19acacffca6a3e74f30997ad06162a70fd2a181ca2e")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-quast/package.py]: These sha256 are no longer correct so need to be updated.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
The change means the old sha256 are not longer correct.

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Missing new deps on joblib/simplejson. While you're at it, it would be good to determine which boost variants are needed. Python/Perl prob don't have the right build types.

**Comment by Pandapip1** [comment]
I don't know how to determine which boost components quast uses. By default, it installs all of them:

https://github.com/ablab/quast/blob/2723558fa28d8a72f8c0cadd4f5f75e1ad6e2e9c/Dockerfile#L5C1-L10

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed all of the listed sha256.

File: var/spack/repos/builtin/packages/py-quast/package.py [hunk 2]
```diff
@@ -27,6 +31,8 @@ class PyQuast(PythonPackage):
     depends_on("python@2.5:,3.3:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-quast/package.py]: Perl and Python are currently build/link deps, meaning they aren't available at run-time. Is this what you want?

**Comment by adamjstewart** [comment]
I also don't know anything about boost. Usually CMakeLists.txt tells you what boost components are required, or maybe they import headers, but I don't see any of that here.

File: var/spack/repos/builtin/packages/py-quast/package.py (hunk 2 above)
Comment (Pandapip1) [reply var/spack/repos/builtin/packages/py-quast/package.py]: What do you mean by that?

**Comment by Pandapip1** [comment]
There is no CMakeLists.txt, and I don't have the time to go through all of the headers looking for boost imports. If the community wants to, they can reduce the list of boost components that are marked as a dependency, but I declare that out-of-scope for this PR.

File: var/spack/repos/builtin/packages/py-quast/package.py (hunk 2 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-quast/package.py]: See https://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-types

**Comment by Pandapip1** [comment]
Any updates?

**Review by tldahlgren — DISMISSED** [review]
Confirmed all of the version sha256 are now correct.

**Comment by adamjstewart** [comment]
Still need to fix dependency types

**Comment by Pandapip1** [comment]
Fixed the dependency types.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**